### ISSUE TITLE
Suspend does not work

### ISSUE THREAD

**Issue by damhiya** [NixOS/nixpkgs#212415]
### Describe the bug
I can't suspend my system with recent nixos-unstable branch. This is the related kernel log.
```
[  147.167036] sof-audio-pci-intel-tgl 0000:00:1f.3: hda_dsp_set_hw_params_upon_resume: failure in hda_dsp_dais_suspend
[  147.167039] sof-audio-pci-intel-tgl 0000:00:1f.3: error: setting hw_params flag during suspend -22
[  147.167040] sof-audio-pci-intel-tgl 0000:00:1f.3: PM: pci_pm_suspend(): snd_sof_suspend+0x0/0x10 [snd_sof] returns -22
[  147.167051] sof-audio-pci-intel-tgl 0000:00:1f.3: PM: dpm_run_callback(): pci_pm_suspend+0x0/0x160 returns -22
[  147.167054] sof-audio-pci-intel-tgl 0000:00:1f.3: PM: failed to suspend async: error -22
[  147.492087] PM: Some devices failed to suspend, or early wake event detected
```

### Steps To Reproduce
1. Install pipewire with recent nixos-unstable branch (1b1f50645af2a70dc93eae18bfd88d330bfbcf7f)
2. run `systemctl suspend`

### Expected behavior
The system should suspend without error.

### Additional context
There is a related report on pipewire: https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/2960
It seems the problem occurs with pipewire 0.3.64

One possible solution is reverting https://github.com/NixOS/nixpkgs/pull/210351 and wait until the upstream fix the bug. Suspending is critical for laptop.

### Notify maintainers
@jtojnar @kranzes @K900 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.19, NixOS, 23.05 (Stoat), 23.05.20230123.1b1f506`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by K900** [comment]
Upstream kernel bug: https://bugzilla.kernel.org/show_bug.cgi?id=216944. I don't think reverting Pipewire is appropriate given the very limited impact.

**Comment by K900** [comment]
If you want a quick solution, you can set 

```nix
services.pipewire.package = pkgs.pipewire.overrideAttrs(old: {
    patches = old.patches ++ [(pkgs.fetchpatch {
        url = "https://gitlab.freedesktop.org/pipewire/pipewire/-/commit/f62c40c205f84d70d41fef3f3d361b0e4651a50e.patch";
        revert = true;
        hash = "sha256-T8OZv9AYFTwDN0JwFHaddsum1YibCpxyxGuhx+Ogg6g=";
    })];
});

**Comment by damhiya** [comment]
Can we apply that patch to nixpkgs? My opinion is pipewire 0.3.64 without fix shouldn't be provided with release branch of NixOS.

Of course it takes time to be released, and the bug might be fixed before next NixOS release.

**Comment by K900** [comment]
Once again, this isn't a Pipewire bug, it's a kernel bug, and only people with specific hardware are affected by it. It's very likely the kernel or the firmware will be fixed to handle this correctly eventually, so I'd rather not diverge from upstream Pipewire for a workaround.

**Comment by damhiya** [comment]
Okay, that is not a pipewire's bug. Sorry for somewhat misleading wording.
I suggested it because it temporarily prevents common users from experiencing the bug, but I respect your decision if you prefer maintainability over it.

**Comment by damhiya** [comment]
After testing, I realized that downgrading pipewire does not solve the problem.
I tested
- reverting https://gitlab.freedesktop.org/pipewire/pipewire/-/commit/f62c40c205f84d70d41fef3f3d361b0e4651a50e.patch
- pipewire 0.3.63
- pipewire 0.3.60

But downgrading NixOS revision to https://github.com/NixOS/nixpkgs/commit/9930b3a9c2a364dde4235ed46664d8fffe932b7c gave me a system that suspend works. The version of pipewire in this revision is 0.3.64.
